bridge_adaptivity/module/models.py

```diff
@@ -235,30 +252,34 @@ def save(self, *args, **kwargs):
         """Extension which sends notification to the Adaptive engine that Activity is created/updated."""
         initial_id = self.id
         if initial_id:
-            if not ENGINE.update_activity(self):
-                raise ValidationError
+            for engine in self.collection.collection_groups.all():
+                if not engine.update_activity(self):
+                    raise ValidationError
```

Comment:
you should raise instance if exception, not a class.
raising exception class were in old python versions, and now you should initialize it.

according to [this](https://stackoverflow.com/questions/16706956/is-there-a-difference-between-raise-exception-and-raise-exception-without) there's no difference between these methods, but in my opinion it would be more clear to raise exception some text message which will clarify why did you raise it.

let's discuss it?

Reply:
It would be removed in the coming PR with the engine synchronization.